### DISCUSSION TITLE
maiko: Fate of NATIVETRAN

### DISCUSSION THREAD

**Discussion by waywardmonkeys** [Interlisp/medley#89] [Implementation history] [answered]
`NATIVETRAN` is described as:

     NATIVETRAN  If true, code to support the native-code translator is
     included in the emulator.  This feature was never released,
     and most such code is probably obsolete.

There's some code involved here, like `src/loader.c` which doesn't compile and hasn't in a while. (If you fix a couple of the missing headers, you quickly run into further changes...)

There's also this from `src/subr.c`:

```
#ifdef NATIVETRAN
    case sb_LOAD_NATIVE_FILE:
      POP_SUBR_ARGS;
      /* to become OBSOLETE */
      {
        TopOfStack = dynamic_load_code(args);
        break;
      }
#endif
```

There's also all of `inc/native.h`, which isn't included or used anywhere.

It looks like this has some complications spread through the code, never worked, never shipped, and even in the old comments, is noted as obsolete.

**Comment by masinter** [comment, marked as answer]
Native Translation (transforming sequence of byte codes to native code) isn't on the path to any of our various goals. The only reason to do it would be performance, which isn't the point and there are lots of ways it wouldn't work out.
Transforming to little-endian and 64-bit pointers would be a lot easier. Nuke it.

**Reply by waywardmonkeys** [reply]
This has been removed on the Maiko side of things.

**Comment by waywardmonkeys** [comment]
FYI: @nbriggs

But this isn't as easy as just removing things that are `#ifdef NATIVETRAN` ... there's a bunch of pieces here and there that aren't covered by an `#ifdef`.

**Comment by nbriggs** [comment]
It's also the case that removing it isn't really on the path to finding the other problems, so while removing the hooks may be a worthy goal in the longer term I don't think it's near the top of any priority list.

**Comment by masinter** [comment]
i converted this from an Issue to a Discussion in the "implementation history" category, following the principle that Issues should be reserved for things that are acdtionable, that someone could self-assign and make progress on, and other things are Discussions of general topics or trying to come up with policy.  At lease that is how I read the intention of the feature.
